### ISSUE TITLE
Custom Loss in tensorflow 2.0

### ISSUE THREAD

**Issue by hagianga21** [tensorflow/tensorflow#32956]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
Hi, I try to write a custom loss in tensorflow 2.0 RC. But the result of customLoss2 is not equal to sparse_categorical_crossentropy. How can I write it the right way? You can easily copy and run the code in Colab.

**Code to reproduce the issue**

```
from __future__ import absolute_import, division, print_function, unicode_literals
import os
import matplotlib.pyplot as plt
import tensorflow as tf
print("TensorFlow version: {}".format(tf.__version__))
print("Eager execution: {}".format(tf.executing_eagerly()))
from tensorflow.keras.layers import Dense, Flatten, Conv2D
from tensorflow.keras import Model

mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])
def customLoss2(y_true, y_pred):
    return tf.reduce_mean(tf.keras.losses.sparse_categorical_crossentropy(y_true, y_pred, from_logits=True))
  
def customLoss3(y_true, y_pred):
    return tf.reduce_mean(tf.keras.losses.sparse_categorical_crossentropy(y_true, y_pred))
  
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy', customLoss2])

model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)
```

**Other info / logs**
TensorFlow version: 2.0.0-rc2
Eager execution: True
Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 5s 85us/sample - loss: 0.3025 - accuracy: 0.9114 - customLoss2: 1.6056
Epoch 2/5
60000/60000 [==============================] - 5s 84us/sample - loss: 0.1438 - accuracy: 0.9574 - customLoss2: 1.5303
Epoch 3/5
60000/60000 [==============================] - 5s 79us/sample - loss: 0.1063 - accuracy: 0.9677 - customLoss2: 1.5126
Epoch 4/5
60000/60000 [==============================] - 5s 78us/sample - loss: 0.0889 - accuracy: 0.9730 - customLoss2: 1.5038
Epoch 5/5
60000/60000 [==============================] - 5s 79us/sample - loss: 0.0738 - accuracy: 0.9769 - customLoss2: 1.4975
10000/1 [=====] - 1s 91us/sample - loss: 0.0398 - accuracy: 0.9763 - customLoss2: 1.4932
[0.07908454576081131, 0.9763, 1.4931514]
